src/frontend/src/optimizer/plan_node/logical_source.rs

```diff
@@ -70,6 +70,9 @@ impl LogicalSource {
         ctx: OptimizerContextRef,
         as_of: Option<AsOf>,
     ) -> Result<Self> {
+        // XXX: should we reorder the columns?
```

Comment:
I think the order does not matter much. The columns field is essentially a map indexed by the column id.

Reply:
I think so, it's just for what users will see in `SELECT *`. 

But I'm not sure if we rely on the position of hidden column like `_row_id` somewhere.. For `projected_row_id` we do so...